### DISCUSSION TITLE
WPF 在样式中给属性(如 Content)加入控件时,只会显示一次

### DISCUSSION THREAD

**Discussion by Hakoyu** [BYJRK/DotNet-Discussions#19] [Q&A] [answered]
**正常情况**
![image](https://user-images.githubusercontent.com/48956359/216829363-3e9f6739-2b06-4260-b716-6e71d65539f0.png)
**使用新控件情况**
![image](https://user-images.githubusercontent.com/48956359/216829335-9c5703f6-4a91-45d1-ad95-69fa8c8afd54.png)

**Comment by BYJRK** [comment]
在，看看 ListBoxItemHelper 的源代码？😂

**Reply by Hakoyu** [reply]
本质上肯定和 ListBoxItemHelper 没关系,不然我也不会问了
![image](https://user-images.githubusercontent.com/48956359/216907126-2f275bc2-16cb-4539-aa1f-44238a302c58.png)

**Comment by Hakoyu** [comment]
应该是实现方式有差别
与`Content`不同 `ContextMenu`是可以做到每个`Item`独立的
甚至不是一个样式,而是一个存在的控件
![image](https://user-images.githubusercontent.com/48956359/217265324-2354193f-67a4-4596-a9c0-796cb92b22a5.png)

**Comment by TruePluto** [comment, marked as answer]
`x:Shared="false"`
真的不知道为什么大家都不愿意把问题先抽取出来做一个可重现的Demo
我觉得你的`style`中设置`Icon`是来自于控件的`Icon` 而这个Icon可能是绑定到某个资源，在VisualTree中资源是能不能共享的，要使用共享资源需要将资源设置为 `x:Shared="false"`
不知道能不能解决您的问题

**Reply by Hakoyu** [reply]
根据 [Learn](https://learn.microsoft.com/zh-cn/dotnet/desktop/xaml-services/xshared-attribute) 的说法
把控件设置为资源项,然后设置`x:Shared="false"`,即可在调用此资源时成一个新控件
尽管在设计视图中还是保持原样,但实际运行是可以正常显示的
**非常感谢你的帮助**
![image](https://user-images.githubusercontent.com/48956359/217464791-d25af15b-11cb-4610-805d-2c3dc36b1c0d.png)
![image](https://user-images.githubusercontent.com/48956359/217465040-f19577ff-6477-4493-8866-76812f621af8.png)

**Reply by TruePluto** [reply]
谢谢您的纠正 我太急于写回复了 
确实应该为` x:Shared="false` 我已经修改回复了，谢谢 很高兴能解决您的问题
